activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb

```diff
@@ -14,14 +14,6 @@ def mysql2_connection(config)
       config[:username] = 'root' if config[:username].nil?
       config[:flags] ||= 0
 
-      if Mysql2::Client.const_defined? :FOUND_ROWS
-        if config[:flags].kind_of? Array
-          config[:flags].push "FOUND_ROWS".freeze
-        else
-          config[:flags] |= Mysql2::Client::FOUND_ROWS
```

Comment:
Only way currently is fork Rails or monkey patch. The trouble with making this configurable, as I see it, is that ActiveRecord may has assumptions about the FOUND_ROWS flag being set and could fail in subtle ways that would be rough to debug.